libs/storage/Tsavorite/cs/src/core/Index/Tsavorite/Implementation/FindRecord.cs

```diff
@@ -130,6 +142,26 @@ private bool TraceBackForKeyMatch(ref TKey key, ref RecordSource<TKey, TValue, T
             return false;
         }
 
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        private bool TraceBackForKeyMatch(ref TKey key, ref RecordSource<TKey, TValue, TStoreFunctions, TAllocator> recSrc, long minAddress, long maxAddress)
+        {
+            // PhysicalAddress must already be populated by callers.
+            ref var recordInfo = ref recSrc.GetInfo();
+            if (IsValidTracebackRecord(recordInfo) && recSrc.LogicalAddress < maxAddress && storeFunctions.KeysEqual(ref key, ref recSrc.GetKey()))
+            {
+                recSrc.SetHasMainLogSrc();
+                return true;
+            }
+
+            recSrc.LogicalAddress = recordInfo.PreviousAddress;
+            if (TraceBackForKeyMatch(ref key, recSrc.LogicalAddress, minAddress, maxAddress, out recSrc.LogicalAddress, out recSrc.PhysicalAddress))
+            {
+                recSrc.SetHasMainLogSrc();
+                return true;
+            }
+            return false;
+        }
+
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
```

Comment:
The duplication saves an "if" but makes the code more confusing. Add some begin/end comments identifying the clones so it's easier to visually parse.